### DISCUSSION TITLE
RE: Lasers and Sculpfun S9 and the T41U5XBB

### DISCUSSION THREAD

**Discussion by silver2row** [grblHAL/core#637] [Show and tell]
Okay and So,

I never used nor understood the power of a diode laser. I always used gas lasers (CO2). Now, my machine is a bit slow for handling lasers as of now. So, I will not be using the Sculpfun S9 diode laser any longer on the CNC machine (way too slow).

I have successfully used the T41U5XBB with Teensy4.1 along with the diode laser to start producing. 

Seth

P.S. I knew CNC worked but I had trouble with the lasers at first. Powerful, i.e. especially when out of focus and way to far away from the landing point (material).

Also...

I think this board is very promising and useful so far along with the 32-bit Teensy4.1. 

In hindsight, I should have known to put the GND on the T41U5XBB and not at the 12v PSU GND. Lasers!

**Comment by silver2row** [comment]
It works out of the box with LaserGRBL too. 

Seth

**Comment by silver2row** [comment]
Has anyone got their laser(s) to work with Lightburn and the T41U5XBB along with the Teensy4.1?

Seth

P.S. If so, maybe a build of sorts would be nice for others trying to do things with lasers with these components.

**Reply by terjeio** [reply]
> Has anyone got their laser(s) to work with Lightburn and the T41U5XBB along with the Teensy4.1?

I believe so, perhaps @phil-barrett know?
BTW the Teensy 4.1 is the fastest processor running grblHAL, with tuning and capable hardware feed rates > 20.000 mm/min is possible when engraving bitmaps.

**Reply by silver2row** [reply]
Thank you...I appreciate you guys replying and steadily providing support.

**Comment by phil-barrett** [comment]
A number of people are using the T41U5XBB with lightburn.  Here is a blog about [using diode lasers](https://www.grbl.org/single-post/using-a-diode-laser-with-grblhal).

For a CO2 laser, the laser PSU may have some specific triggering requirements (trigger low, for example). Best to ask specific questions about that.

**Reply by silver2row** [reply]
I will look into the diode laser ideas for grbl...

Currently, my machine for the diode laser is way too slow. I cut a quarter inch of wood in seconds with only one pass. I was extremely surprised about the strength of the diodes in the laser. 

All I really did was use the CNC machine, lead screw type machine, to handle the laser instead of the spindle with a couple of changes in the software in ioSender.

12v, PWM-GND, and diode laser and I was in business. I used to use only gas (CO2) lasers but these diode lasers are remarkably inexpensive and sell. I have a Sculpfun laser now (diode). 

Anyway, I have another type of controller, BeagleBone Green derived, that handles the CO2. I have yet to put it together just yet. 

Seth

P.S. Thank you both for supporting the board and people that purchase your products. Outside of compliments, I noticed in LaserGRBL, there is fictionally no way to handle the Z_Axis with success or at least I was unable to do so with my knowledge of the source.

**Reply by phil-barrett** [reply]
Yeah, most laser engravers don't have Z motor unless it is for manually raising/lowering the bed. I think most people with CO2 lasers don't want to raise/lower the focus lens. But with a diode, being able to control Z allows allows for height mapping so the laser can engrave uneven surfaces.

**Reply by silver2row** [reply]
Right. That is what exactly I have come to understand. It is neat that the source can alter PWM "power" to see when/where to alleviate some of the current to handle height mapping. 

I am going to test some more... If you guys/gals need any testing done, let me know. So far, the T41U5XBB with the PWM onboard handles a "threshold" of sorts and this is mostly from my lack of testing. I currently have been unable to make the machine go faster (lead screws) and I think this "threshold" is due to not having belts. It is really just a slow machine.

I tested some less expensive builds of belts on motors. This was a mistake. The less expensive builds I found did not handle belt action like I would have liked.

And right about CO2 lasers not needing to get out of focus with the set-in-stone builds for the Z-Axis. For instance, my H-Series from fslaser, which is a very old model from them, using gas and light along with liquid cooling. The only adjustments for this CO2 laser that needs to be made can be done by hand. The beam travels left parallel to the cartesian plane, forward to the front-left corner, and then to the "Z_Axis" which then goes straight down with another lens. There is an actual red beam laser that shows where, in theory, the focal point is located. 

Of course this needs correction every so often. The lenses can be manipulated, i.e. thus leaving an error with many uses after so much movement. And get this idea...

1. The company, whomever put the machine together, sent me a lemon.
2. And by lemon, it did not work.
3. I had to deal with their company of guessers on what the "mechanics" and assembly techs did to put together my machine.

Finally, we figured out, I really did the `work` with time and effort, that the machine was not working due to the wiring and "safety" mechanism on the box. A simple "does not work" attached door sensor (NC) was wired incorrectly and not placed where it should have been placed. Also, they used many washers that were unneeded on many sections of the belt driven configuration. Shorter screws would have done...

At the time with their company, I am guessing people were looking for freebies and stealing. They stole from the company, lied to get the jobs most obviously, and then did a disservice to the customer (me). 

Seth

P.S. I got it to work in the end and then an "idiot" in my life messed with the wiring without my knowledge. It took me taking apart the entire machine, wiring and all, to figure out they placed the PVC coating of the wiring in the Stellaris daughter card screw terminals. 

One little PVC coated wire without wiring attached in the screw terminal and poof. Just like that idea, it stopped firing. 

```
Aw!
```

By that point in time, I gave up and called it quits for a bit. I dropped people visiting and workers doing work on the ole homestead. People steal. Blah...it sucks but it is part of life. Now that I want back in, I want to learn more again. I am a user too but I like learning more than just being a utilizer of sorts. 

I was using the beaglebone black from beagleboard.org for a while to learn about 32-bit infrastructure and programming. The PRUs onboard have been known to work at or around 200Mhz depending on how exactly they are utilized. I am not a seller or major contributor of their builds and use cases but I have learned to enjoy time on Earth and building various appliances with their boards. 

I first started with home security and notifications via email from when doors and other things have been opened or closed... NC mostly.

Anyway, enough blabbing. I hope you guys have a nice Christmas and New Years... I will try to grab another board soon to use it with my upcoming build. Maybe I can get the belts out and move a bit faster...
